src/SignalR/common/Http.Connections/src/Internal/HttpConnectionDispatcher.cs

```diff
@@ -331,7 +331,8 @@ private async Task ExecuteAsync(HttpContext context, ConnectionDelegate connecti
                         // Mark the connection as inactive
                         connection.LastSeenUtc = DateTime.UtcNow;
 
-                        connection.Status = HttpConnectionStatus.Inactive;
+                        // This is done outside a lock because the next poll might be waiting in the lock already and waiting for currentRequestTcs to complete
+                        Interlocked.CompareExchange(ref connection._status, (int)HttpConnectionStatus.Inactive, (int)HttpConnectionStatus.Active);
```

Comment:
So this only prevents moving from the Disposed to Inactive state, right? Is this by itself enough to fixe the "InvalidOperationException: Reading is not allowed after reader was completed."? Are we sure there's no other way to transition out of the Disposed state?

I think a comment discussing what state transition we're trying to prevent here would be nice.